code/crates/core-driver/src/driver.rs

```diff
@@ -197,16 +204,24 @@ where
     }
 
     /// Get a commit certificate for the given round and value id.
-    pub fn get_certificate(
+    pub fn get_commit_certificate(
         &self,
         round: Round,
         value_id: ValueId<Ctx>,
     ) -> Option<&CommitCertificate<Ctx>> {
-        self.certificates
+        self.commit_certificates
             .iter()
             .find(|c| c.round == round && c.value_id == value_id)
     }
 
+    /// Get a polka certificate for the given round.
+    pub fn get_polka_certificate(&self, round: Round) -> Option<&PolkaCertificate<Ctx>> {
+        self.polka_certificates
+            .iter()
+            .filter(|c| c.round <= round)
+            .last()
```

Comment:
Here I did something similar as we do for commit certificates but it feels very hacky and I am not convinced it is right. Will check the Quint spec again to see whether or not this is correct.